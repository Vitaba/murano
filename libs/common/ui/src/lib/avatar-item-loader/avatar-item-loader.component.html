<ngx-content-loading
[height]="height"
[width]="width"
[speed]="2"
[primaryColor]="'#f3f3f3'"
[secondaryColor]="'#ecebeb'">
  <svg:g ngx-rect x="70" y="15" rx="4" ry="4" width="117" height="6"></svg:g>
  <svg:g ngx-rect x="70" y="35" rx="3" ry="3" width="85" height="6"></svg:g> 
  <svg:g ngx-circle cx="30" cy="30" r="30"></svg:g>
</ngx-content-loading>