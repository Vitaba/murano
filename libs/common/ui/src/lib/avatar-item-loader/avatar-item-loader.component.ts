import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnChanges, SimpleChanges } from '@angular/core';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'vitaba-avatar-item-loader',
  styleUrls: ['./avatar-item-loader.component.scss'],
  templateUrl: './avatar-item-loader.component.html',
})
export class AvatarItemLoaderComponent implements OnChanges {
  @Input() public width;
  @Input() public height;

  public constructor(
    private readonly _changeDetectionRef: ChangeDetectorRef,
  ) {}

  public ngOnChanges(changes: SimpleChanges): void {
    const currentValue =
    changes.data.currentValue as { height: number; width: number };

    this.width = currentValue.width;
    this.height = currentValue.height;
    this._changeDetectionRef.detectChanges();
  }
}
