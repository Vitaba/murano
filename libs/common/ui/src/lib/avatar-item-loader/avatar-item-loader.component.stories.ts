import { ContentLoaderModule } from '@vitaba/content-loader';
import { number, object } from '@storybook/addon-knobs';
import { AvatarItemLoaderComponent } from './avatar-item-loader.component';

// tslint:disable-next-line:no-default-export
export default {
  title: 'AvatarItemLoaderComponent',
};

export const primary = () => ({
  component: AvatarItemLoaderComponent,
  moduleMetadata: {
    declarations: [AvatarItemLoaderComponent],
    imports: [ContentLoaderModule],
  },
  props: {
    data: {
      colors: object('colors', {
        primary: '#f4f4f4',
        secondary: '#ecebeb',
      }),
      height: number('height', 100),
      speed: number('speed', 2),
      width: number('width', 1000),
    },
  },
});
