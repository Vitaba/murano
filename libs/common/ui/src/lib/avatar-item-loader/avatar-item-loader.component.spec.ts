import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AvatarItemLoaderComponent } from './avatar-item-loader.component';

describe('AvatarItemLoaderComponent', () => {
  let component: AvatarItemLoaderComponent;
  let fixture: ComponentFixture<AvatarItemLoaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AvatarItemLoaderComponent],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AvatarItemLoaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
