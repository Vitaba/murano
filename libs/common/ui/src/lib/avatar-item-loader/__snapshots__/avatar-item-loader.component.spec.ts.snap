// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarItemLoaderComponent should create loader 1`] = `
<vitaba-avatar-item-loader
  _changeDetectionRef={[Function ViewRef_]}
  colors={[Function Object]}
  height={[Function Number]}
  idClip={[Function String]}
  idGradient={[Function String]}
  speed={[Function Number]}
  width={[Function Number]}
>
  <content-loader
    ng-reflect-height="100"
    ng-reflect-id-clip="avatar-item-loader-idclip"
    ng-reflect-id-gradient="avatar-item-loader-gradient"
    ng-reflect-ignore-base-url="true"
    ng-reflect-primary-color="#f4f4f4"
    ng-reflect-secondary-color="cecece"
    ng-reflect-speed="2"
    ng-reflect-width="100"
  >
    <svg
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 100 100"
    >
      <rect
        clip-path="url(#avatar-item-loader-idclip)"
        height="100"
        ng-reflect-ng-style="[object Object]"
        style="fill: url(#avatar-item-loader-gradient);"
        width="100"
        x="0"
        y="0"
      />
      <defs>
        <clippath
          id="avatar-item-loader-idclip"
        >
          <rect
            height="6"
            rx="4"
            ry="4"
            width="117"
            x="70"
            y="15"
          />
          <rect
            height="6"
            rx="3"
            ry="3"
            width="85"
            x="70"
            y="35"
          />
          <circle
            cx="30"
            cy="30"
            r="30"
          />
        </clippath>
        <lineargradient
          id="avatar-item-loader-gradient"
        >
          <stop
            offset="0%"
            stop-color="#f4f4f4"
            stop-opacity="1"
          >
            
            
            <animate
              attributeName="offset"
              dur="2"
              repeatCount="indefinite"
              s=""
              values="-3; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="cecece"
            stop-opacity="1"
          >
            
            
            <animate
              attributeName="offset"
              dur="2"
              repeatCount="indefinite"
              s=""
              values="-2; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#f4f4f4"
            stop-opacity="1"
          >
            
            
            <animate
              attributeName="offset"
              dur="2"
              repeatCount="indefinite"
              s=""
              values="-1; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
  </content-loader>
</vitaba-avatar-item-loader>
`;
